db/compaction/compaction_picker_level.cc

```diff
@@ -925,11 +925,16 @@ bool LevelCompactionBuilder::PickSizeBasedIntraL0Compaction() {
   }
   uint64_t l0_size = 0;
   for (const auto& file : l0_files) {
-    l0_size += file->fd.GetFileSize();
+    assert(file->compensated_file_size >= file->fd.GetFileSize());
+    // Compact down L0s with more deletions.
+    l0_size += file->compensated_file_size;
   }
+
+  // Avoid L0->Lbase compactions that are inefficient for write-amp.
   const uint64_t min_lbase_size =
-      l0_size * static_cast<uint64_t>(std::max(
-                    10.0, mutable_cf_options_.max_bytes_for_level_multiplier));
+      l0_size *
+      static_cast<uint64_t>(std::max(
+          10.0, mutable_cf_options_.max_bytes_for_level_multiplier * 2));
```

Comment:
Hi I need some context here to help with the review. 

> ...Also scale down the limit on total L0 size further to be more cautious.

Is this referring to increasing min_lbase_size here? 
- If so, how are these two related? 
-Why does increasing min_lbase_size "avoid L0->Lbase compactions that are inefficient for write-amp"? I could only see the code below is more easily to be executed so it's more likely we don't do PickSizeBasedIntraL0Compaction. But how L0->Lbase is impacted??
```
if (lbase_size <= min_lbase_size) {
    return false;
  }
```

Reply:
Sorry for the confusion. The comment is not related the change on min_lbase_size, it's more about why we have a limit on the lbase size. 

For "scale down the limit on total L0 size further to be more cautious", your understanding is correct. It makes the requirement stricter for intra-L0 compaction. The motivation is to avoid accumulating too much data in L0, especially if L0 has many deletion.

Reply:
Thanks - might worth updating the PR summary to be more explicit just to help future readers that aren't super familiar with compactions.